### PR TITLE
package-indexer: security enhancements

### PR DESCRIPTION
**Pull request by alltilla**
We are planning to move the package indexing to GitHub Actions. With this PR I am adding some security enhancements to the indexing script.

Namely:
  - Loading the config content through a CLI with `--config-content`.
  - Hiding YAML loading errors, so we do not leak any part of the config in case of a load failure.
  - Cleaning up the temporary `GNUPGHOME` dir after it is no longer in use.
  - Moving the temporary `GNUPGHOME` dir to the script's current dir.
  - Supporting passphrase protected GPG keys, with the passhprase loaded from `STDIN`.

Note, that until we start indexing the packages in GitHub Actions, these changes do not get used anywhere.
I tested the changes locally with both passphrase protected and not protected GPG keys, if you need help to try the new code out, I am happy to help.

Signed-off-by: Attila Szakacs <attila.szakacs@oneidentity.com>

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by alltilla** [comment]
The disk-buffer unit test failure is caused by another problem.

**Comment by MrAnno** [comment]
The disk-buffer test issue is fixed in #4004.

**Comment by alltilla** [comment]
Rebased to master.

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: packaging/package-indexer/index-packages.py [hunk 1]
```diff
@@ -125,6 +131,8 @@ def construct_indexers(cfg: Config, args: dict) -> List[Indexer]:
     indexed_remote_storage_synchronizer = cfg.create_indexed_remote_storage_synchronizer(suite)
     cdn = cfg.create_cdn(suite)
 
+    gpg_key_passphrase = stdin.read() if args["gpg_key_passphrase_from_stdin"] else None
```
Comment (MrAnno) [inline packaging/package-indexer/index-packages.py:134]: We could ask(prompt) for the passphrase in this case and stop reading after the first newline.

**Review by MrAnno — APPROVED** [review]
LGTM.

File: packaging/package-indexer/index-packages.py (hunk 1 above)
Comment (alltilla) [reply packaging/package-indexer/index-packages.py:134]: Good idea, but I would leave that for the future, when the need arises.

**Review by OverOrion — APPROVED** [review]

File: packaging/package-indexer/indexer/deb_indexer.py [hunk 2]
```diff
@@ -122,12 +124,28 @@ def __init__(
             apt_conf_file_path=Path(CURRENT_DIR, "apt_conf", "stable.conf"),
         )
 
+    def __add_gpg_passphrase_params_if_needed(self, command: list) -> list:
+        if self.__gpg_key_passphrase is None:
+            return command
+
+        assert command[0] == "gpg"
+
+        return [
```
[the remark this answers is not in the document]
Comment (alltilla) [reply packaging/package-indexer/indexer/deb_indexer.py:143]: Sure, added them! :)

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Review by OverOrion — APPROVED** [review]
LGTM!

**Review by MrAnno — APPROVED** [review]

**Pull request merged**